File: src/main/java/com/koddy/server/notification/domain/repository/query/response/NotificationDetails.java
package com.koddy.server.notification.domain.repository.query.response;

import com.koddy.server.coffeechat.domain.model.CoffeeChatStatus;
import com.koddy.server.coffeechat.domain.model.Reason;
import com.koddy.server.coffeechat.domain.model.Reservation;
import com.querydsl.core.annotations.QueryProjection;

import java.time.LocalDateTime;

public record NotificationDetails(
        long id,
        boolean read,
        LocalDateTime createdAt,
        long memberId,
        String memberName,
        String memberProfileImageUrl,
        long coffeeChatId,
        CoffeeChatStatus coffeeChatStatus,
        Reason coffeeChatReason,
        Reservation coffeeChatReservation
) {
    @QueryProjection
    public NotificationDetails {
    }
}

File: src/main/kotlin/com/koddy/server/notification/domain/repository/query/response/NotificationDetails.kt
package com.koddy.server.notification.domain.repository.query.response

import com.koddy.server.coffeechat.domain.model.Reason
import com.koddy.server.coffeechat.domain.model.Reservation
import com.koddy.server.notification.domain.model.NotificationType
import java.time.LocalDateTime

data class NotificationDetails(
    val id: Long,
    val isRead: Boolean,
    val coffeeChatStatusSnapshot: String,
    val type: NotificationType,
    val createdAt: LocalDateTime,
    val memberId: Long,
    val memberName: String,
    val memberProfileImageUrl: String?,
    val coffeeChatId: Long,
    val coffeeChatReason: Reason?,
    val coffeeChatReservation: Reservation?,
)
